maploc/utils/wrappers.py

```diff
@@ -203,7 +206,118 @@ def magnitude(self) -> Tuple[torch.Tensor]:
         return dr, dt
 
     def __repr__(self):
-        return f"Pose: {self.shape} {self.dtype} {self.device}"
+        return f"Transform3D: {self.shape} {self.dtype} {self.device}"
+
+
+class Transform2D(TensorWrapper):
+    """SE(2) transformation with 2-DoF translation and 1-DoF rotation"""
+
+    def __init__(self, data: torch.Tensor):
+        assert data.shape[-1] == 3  # angle_deg, x, y
+        super().__init__(data)
+
+    @property
+    def angle(self) -> torch.Tensor:
+        """Returns angle in degrees"""
+        return self._data[..., :1]
+
+    @property
+    def t(self) -> torch.Tensor:
+        """Underlying translation vector with shape (..., 2)."""
+        return self._data[..., -2:]
+
+    @classmethod
+    def from_degrees(cls, angle: torch.Tensor, t: torch.Tensor):
+        """SE(2) pose from degrees. Rotation is counter-clockwise from +ve X"""
+        rt_flat = torch.cat([angle, t[..., 0:2]], -1)
+        return cls(rt_flat)
+
+    @classmethod
+    @autocast
+    def from_Rt(cls, R: torch.Tensor, t: torch.Tensor):
+        """Pose from a 2D rotation matrix and 2D translation vector.
+        Accepts numpy arrays or PyTorch tensors.
+
+        Args:
+            R: rotation matrix with shape (..., 2, 2).
+            t: translation vector with shape (..., 2).
+        """
+        assert R.shape[-2:] == (2, 2)
+        assert t.shape[-1] == 2
+        assert R.shape[:-2] == t.shape[:-1]
+        angle_deg = torch.arctan2(R[..., 1, 0], R[..., 0, 0]) * 180 / math.pi
+        return cls.from_degrees(angle_deg[None], t[..., :2])
+
+    @classmethod
+    def from_Transform3D(cls, transform: "Transform3D"):
```

Comment:
The name is identical to SNAP's `from_Transform3D` but the behavior is very different. Please fix, e.g. by renaming to `camera_2d_from_3d`